src/cpu/cpu.cpp

```diff
@@ -2261,87 +2261,72 @@ class CPU final : public Module_base {
 		CPU_Cycles=0;
 		CPU_SkipCycleAutoAdjust=false;
 
-		Prop_multival* p = section->Get_multival("cycles");
+		// Sets the value if the string in within the min and max values
+		auto set_if_in_range = [](const std::string &str,
+		                          int &value,
+		                          const int min_value = 0,
+		                          const int max_value = -1) {
+			std::istringstream stream(str);
+			int v = 0;
+			stream >> v;
+			const bool within_min = (v > min_value);
+			const bool within_max = (max_value == -1 || v <= max_value);
+			if (within_min && within_max)
+				value = v;
+		};
+
+		Prop_multival *p = section->Get_multival("cycles");
 		std::string type = p->GetSection()->Get_string("type");
-		std::string str ;
-		CommandLine cmd(0,p->GetSection()->Get_string("parameters"));
-		if (type=="max") {
-			CPU_CycleMax=0;
-			CPU_CyclePercUsed=100;
-			CPU_CycleAutoAdjust=true;
-			CPU_CycleLimit=-1;
-			for (Bitu cmdnum=1; cmdnum<=cmd.GetCount(); cmdnum++) {
-				if (cmd.FindCommand(cmdnum,str)) {
-					if (str.find('%')==str.length()-1) {
+		std::string str;
+		CommandLine cmd(0, p->GetSection()->Get_string("parameters"));
+		if (type == "max") {
+			CPU_CycleMax = 0;
+			CPU_CyclePercUsed = 100;
+			CPU_CycleAutoAdjust = true;
+			CPU_CycleLimit = -1;
+			for (auto cmdnum = 1; cmdnum <= cmd.GetCount(); ++cmdnum) {
+				if (cmd.FindCommand(cmdnum, str)) {
+					if (str.find('%') == str.length() - 1) {
```

Comment:
This checks if the percent character is the last one in the string; use .back for that.